Reviews

Christiandecolombia opened the pull request:
Created functionality for review
controller, repo, service, model. 
the front-end will be completed in the next story.

katame13 committed:
created review schema, entity, service, repo, and worked on the seeder and the tile

katame13 committed:
Merge branch 'main' into reviews

Christiandecolombia committed:
created review controller

Christiandecolombia committed:
Formatted controller file

safang99 commented on src/main/frontend/components/ReviewTile.js:
```diff
@@ -0,0 +1,14 @@
+    </div>
+  )
+}
+
```
Looks like this needs a component which is using the `ReviewTile`

safang99 commented on src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java:
```diff
@@ -0,0 +1,42 @@
+    persistedReview.put("review", review);
+
+    return persistedReview;
+  }
```
I see the `POST` endpoint has been added here.  Is the goal to add a form on the front end?

fbraese replied:
Yes, this is for a form.

safang99 commented on src/main/java/com/launchacademy/reviews/models/Review.java:
```diff
@@ -0,0 +1,47 @@
+//  @ManyToOne
+//  @JsonIgnoreProperties("reviews")
+//  @JoinColumn(name = "site_id")
+//  private Site site;
```
It looks like we will want to bring this back in if there is a one to many relationship to establish between `sites`

fbraese replied:
Yes, this is true & has been brought back.

safang99 commented on src/main/java/com/launchacademy/reviews/seeders/ReviewSeeder.java:
```diff
@@ -0,0 +1,21 @@
+  public void seed(){
+    if(reviewService.findAll().isEmpty()){
+      Site site = siteService.findByName("").get(0);
+    }
```
After finding a site we can add some reviews to it

fbraese replied:
Yes, our next story is to add a form component on each site show page to add reviews.  This story is just to seed and display reviews.

safang99 reviewed (changes requested):
I know you mentioned that this is just the backend and you are building out the frontend in the next PR, but I would like to see the complete feature in one PR unless there is a reason you are trying to get this merged now.  I am a little concerned of the scope of this PR due to the pieces I am seeing so let's sync on the story you are working on

katame13 committed:
worked on site descriptions

added V3.1 scheema
updated seeder, entity, and show page for descriptions
created empty react page for show
created api endpoint for show
added method to site service for optional find by id method

Christiandecolombia committed:
added NewReviewShowPage

Christiandecolombia committed:
Merge branch 'main' into reviews

katame13 committed:
having issues with displaying the show page

posting for help debuging

Christiandecolombia committed:
Added to models

Christiandecolombia committed:
Bugg fixing Bundle.js error

fbraese committed:
Merge branch 'Story3SiteShowPage' of https://github.com/katame13/RunTimeTerror into reviews
We need the code that was being worked on for reviews on the Story 3 branch yesterday

Christiandecolombia committed:
Finished review seeder

Christiandecolombia committed:
Merge branch 'main' into reviews

Christiandecolombia committed:
Finished Site show page with reviews. Completed review feature

Christiandecolombia commented:
Finished Review feature.
Reviews are seeded and displaying on the site show page.

safang99 commented on src/main/frontend/components/NewReviewForm.js:
```diff
@@ -0,0 +1,90 @@
+      </form>
+  )
+}
+export default NewReviewForm
```
Indentation looks to be at a size of 4, but we will want to use 2 to be consistent with the rest of the code.

Also for readability we will want to use an empty line to separate the end of one function from the start of the next.  For example 29&30, 36&37, etc.

Christiandecolombia replied:
Spacing is odd, We use Intellj to format the document.

Christiandecolombia replied:
Updated setting to 2 spaces

safang99 commented on src/main/frontend/components/SitesIndex.js:
```diff
@@ -8,6 +8,7 @@ const SitesIndex = props => {
   const fetchSites = async () => {
     try {
       const response = await fetch("api/v1/sites")
+      console.log("inside index fetch")
```
We can remove this log statement

Christiandecolombia replied:
removed

safang99 commented on src/main/frontend/components/SitesIndex.js:
```diff
@@ -34,6 +35,7 @@ const SitesIndex = props => {
 
   return (
     <div>
+      <p>this is site index</p>
```
I might go with something different for the title of this page.  Index/Show/New may mean alot to us, but it won't mean much to the users visiting your website.

fbraese replied:
Great note!  We meant to update this and just have.  Thank you

safang99 commented on src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java:
```diff
@@ -0,0 +1,42 @@
+//    persistedReview.put("review", review);
+//
+//    return persistedReview;
+//  }
```
If the form is currently unfinished and this is a PR just for displaying reviews, then I would remove the commented out code and the review form from this PR to keep it scoped properly

Christiandecolombia replied:
Commented out code has been removed

safang99 reviewed (changes requested):
Great job and it looks like there is one complete PR in here along with part of a second.  Currently I see a PR for creating reviews and having them display on a `site` show page.  There also seems to be part of adding a `reviews form` which should be it's own PR.  Other than that just minor formatting issues.

Christiandecolombia committed:
Fixed code styling issues.

Christiandecolombia committed:
Removed newReviewForm
